src/GraphControl/Control/Grapher.cpp

```diff
@@ -254,17 +254,22 @@ namespace GraphControl
                 vector<Equation ^> equationVector;
                 equationVector.push_back(equation);
                 UpdateGraphOptions(graph->GetOptions(), equationVector);
-
-                if (analyzer->CanFunctionAnalysisBePerformed())
+                bool variableIsNotX;
+                if (analyzer->CanFunctionAnalysisBePerformed(variableIsNotX))
                 {
                     if (S_OK
                         == analyzer->PerformFunctionAnalysis(
-                            (Graphing::Analyzer::NativeAnalysisType)Graphing::Analyzer::PerformAnalysisType::PerformAnalysisType_All))
+                            (Graphing::Analyzer::NativeAnalysisType)Graphing::Analyzer::PerformAnalysisType::PerformAnalysisType_All)
+                        && !variableIsNotX)
```

Comment:
Shouldn't this be moved up to be included in the if statement above?

```
                if (analyzer->CanFunctionAnalysisBePerformed(variableIsNotX) && !variableIsNotX)
                {
                    if (S_OK
                        == analyzer->PerformFunctionAnalysis(
                            (Graphing::Analyzer::NativeAnalysisType)Graphing::Analyzer::PerformAnalysisType::PerformAnalysisType_All))
                            (Graphing::Analyzer::NativeAnalysisType)Graphing::Analyzer::PerformAnalysisType::PerformAnalysisType_All))
```

That way if the API returns false it will correctly fall down into the block to call KeyGraphFeaturesInfo::Create(CalculatorApp::AnalysisErrorType::VariableIsNotX);